### PR TITLE
CMR-9207: Added legacy labeling to asset thumbnails

### PR DESCRIPTION
**Pull request by dpesall**
In order to support certain clients running scripts against STAC, we need to support the legacy labeling strategy of using the second to last value in a thumbnail URL as a key for the users to scrape for the data.

### CODE REVIEW

File: src/domains/stac.ts [hunk 1]
```diff
@@ -87,12 +88,13 @@ const thumbnailAssets = (concept: Collection | Granule) => {
...
     .reduce((metadataAssets, relatedUrl, idx, available) => {
       const [entry, display] = available.length > 1 ? [`_${idx}`, ` [${idx}]`] : ["", ""];
-
+      const relatedUrlKey = extractAssetMapKey(relatedUrl.url);
       const thumbnailAsset: AssetLinks = {};
       thumbnailAsset[`thumbnail${entry}`] = {
         href: relatedUrl.url,
         title: `Thumbnail${display}`,
         description: relatedUrl.description,
+        [relatedUrlKey]: relatedUrl.url,
```
Comment (jaybarra) [inline src/domains/stac.ts:97]: Should be sure to update one of the granule files to add a related url to catch this in tests.

Comment (dpesall) [reply src/domains/stac.ts:95]: Updated respective tests

File: src/utils/index.ts [hunk 2]
```diff
@@ -42,6 +42,16 @@ export const wrapErrorHandler = (fn: (rq: Request, rs: Response) => Promise<void
...
 };
 
+/**
+ * Returns second to last value in a relatedUrl
+ * to be used as a key for thumbnail assets link.
+ * Defaults to string 'key' in the case of an unexpected URL format.
+ */
+export const extractAssetMapKey = (relatedUrl: string) => {
+  const urlArray = relatedUrl.split(".");
+  return urlArray[urlArray.length - 2] ? urlArray[urlArray.length - 2] : "key";
```
Comment (eudoroolivares2016) [inline src/utils/index.ts:52]: add the test for the case where it has to default to "key"

**Review by eudoroolivares2016 — APPROVED** [review]

File: src/utils/index.ts (hunk 2 above)
Comment (dpesall) [reply src/utils/index.ts:52]: Updated tests to verify this code is working as intended

**Review by macrouch — APPROVED** [review]

**Pull request merged**